src/nvim/shada.c

```diff
@@ -3990,8 +3990,7 @@ static ShaDaReadResult shada_read_next_item(ShaDaReadDef *const sd_reader,
           .data = unpacked.data.via.array.ptr[i],
         };
         {
-          msgpack_unpacked unpacked = unpacked_2;
-          if (unpacked.data.type != MSGPACK_OBJECT_MAP) {
+          if (unpacked_2.data.type != MSGPACK_OBJECT_MAP) {
```

Comment:
I don't see why `unpacked = unpacked_2` would be needed, so just use `unpacked_2` directly.